src/transaction.rs

```diff
@@ -17,8 +17,8 @@ pub struct Instruction {
     /// The program code that executes this transaction is identified by the program_id.
     /// this is an offset into the Transaction::program_ids field
     pub program_ids_index: u8,
-    /// Indices into the keys array of which accounts to load
-    pub accounts: Vec<u8>,
+    /// Indices into the keys array of which accounts to load and whether it's signed.
+    pub accounts: Vec<(u8, bool)>,
```

Comment:
Why does the Transaction sender provide this flag?  "Yeah trust me I signed this Transaction with this key".  The bank should set this flag internally after it unpacks the wire Transaction and sigverifies it

Reply:
Yeah, good call